pkg/framework/objects/service_token/data_source.go

```diff
@@ -0,0 +1,138 @@
+	"github.com/hashicorp/terraform-plugin-framework/types"
+)
+
+var (
+	_ datasource.DataSource              = &serviceTokenDataSource{}
+	_ datasource.DataSourceWithConfigure = &serviceTokenDataSource{}
+)
+
+func ServiceTokenDataSource() datasource.DataSource {
+	return &serviceTokenDataSource{}
+}
+
+type serviceTokenDataSource struct {
+	client *dbt_cloud.Client
+}
+
+// Metadata implements datasource.DataSource.
+func (st *serviceTokenDataSource) Metadata(_ context.Context, req datasource.MetadataRequest, resp *datasource.MetadataResponse) {
+	resp.TypeName = req.ProviderTypeName + "_service_token"
+}
+
+// Configure implements datasource.DataSourceWithConfigure.
+func (st *serviceTokenDataSource) Configure(ctx context.Context, req datasource.ConfigureRequest, resp *datasource.ConfigureResponse) {
+	switch c := req.ProviderData.(type) {
+	case nil: // do nothing
+	case *dbt_cloud.Client:
+		st.client = c
+	default:
+		resp.Diagnostics.AddError("Missing client", "A client is required to configure the service token resource")
+	}
+}
+
+// Schema implements datasource.DataSource.
+func (st *serviceTokenDataSource) Schema(ctx context.Context, req datasource.SchemaRequest, resp *datasource.SchemaResponse) {
+	resp.Schema = schema.Schema{
+		Attributes: map[string]schema.Attribute{
+			"service_token_id": schema.Int64Attribute{
+				Required:    true,
+				Description: "The ID of the service token",
+				// TODO(cwalden): Deprecate this in favor of `id`
+				// DeprecationMessage: "Use `id` instead",
+			},
+			// TODO(cwalden): use Int64Attribute after deprecating `service_token_id`
+			"id": schema.StringAttribute{
+				Computed:    true,
+				Description: "The ID of the service token",
+			},
+			"uid": schema.StringAttribute{
+				Description: "Service token UID (part of the token)",
+				Computed:    true,
+			},
+			"name": schema.StringAttribute{
+				Description: "Service token name",
+				Computed:    true,
+			},
+		},
+		Blocks: map[string]schema.Block{
+			"service_token_permissions": schema.SetNestedBlock{
+				Description: "Permissions set for the service token",
+				NestedObject: schema.NestedBlockObject{
+					Attributes: map[string]schema.Attribute{
+						"permission_set": schema.StringAttribute{
+							Description: "Set of permissions to apply",
+							Computed:    true,
+						},
+						"all_projects": schema.BoolAttribute{
+							Description: "Whether or not to apply this permission to all projects for this service token",
+							Computed:    true,
+						},
+						// TODO(cwalden): Would this be better as a Set of Int64?
+						// TODO(cwalden): Add a validator to ensure that the project ID is set if `all_projects` is false
+						"project_id": schema.Int64Attribute{
+							Description: "Project ID to apply this permission to for this service token",
+							Computed:    true,
+						},
+						// TODO(cwalden): Add validator to ensure that this is configurable for the given `permission_set`
```

Comment:
Same comment as for the `resource` side.

Reply:
Same response as for the `resource` side 😉